examples/angular/basic/src/basic-example.component.ts

```diff
@@ -0,0 +1,150 @@
+import { AngularQueryDevtoolsComponent } from '@tanstack/angular-query-devtools-experimental'
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+  signal,
+} from '@angular/core'
+import { CommonModule, NgIf } from '@angular/common'
+import {
+  injectQuery,
+  injectQueryClient,
+} from '@tanstack/angular-query-experimental'
+import axios from 'axios'
+
+type Post = {
+  id: number
+  title: string
+  body: string
+}
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  selector: 'post',
+  standalone: true,
+  imports: [NgIf],
+  template: `
+    <div>
+      <div>
+        <a (click)="setPostId.emit(-1)" href="#"> Back </a>
+      </div>
+      <ng-container *ngIf="postQuery().status === 'pending'">
+        Loading...
+      </ng-container>
+      <ng-container *ngIf="postQuery().status === 'error'">
+        Error: {{ postQuery().error?.message }}
+      </ng-container>
+      <ng-container *ngIf="postQuery().data as post">
+        <h1>{{ post.title }}</h1>
+        <div>
+          <p>{{ post.body }}</p>
+        </div>
+        <div *ngIf="postQuery().isFetching">Background Updating...</div>
+      </ng-container>
+    </div>
+  `,
+})
+export class PostComponent {
+  @Output() setPostId = new EventEmitter<number>()
+
+  // Until Angular supports signal-based inputs, we have to set a signal
+  @Input({ required: true })
+  set postId(value: number) {
+    this.postIdSignal.set(value)
+  }
+  postIdSignal = signal<number>(0)
+  postQuery = injectQuery(() => ({
+    enabled: this.postIdSignal() > 0,
+    queryKey: ['post', this.postIdSignal()],
```

Comment:
Should we add the possibility to pass an observable into the `queryKey`? Great things are now available with signals, but we still have observables.

Reply:
for example:

```typescript
queryKey: ['post', this.postId$]
// or
queryKey: postId$.pipe(map(id => ['post', id]))
```

Reply:
Also other props like `enabled` might be reactive. Probably `injectQuery` itself should be able to receive `Observable`

Reply:
Hi, what do you think about  using [toSignal()](https://angular.dev/guide/signals/rxjs-interop#tosignal)?

`queryKey: ['post', toSignal(this.postId$)]`

Would that be sufficient? Willing to look into adding first class support for observables if many people want it but on the other hand it might increase complexity quite a bit and Angular has good interop between observables and signals.

Reply:
Yes, it might suffice, but it needs to be tested in real situations for potential leaks or other issues.